includes/sanitizers/class-amp-base-sanitizer.php

```diff
@@ -603,6 +603,28 @@ public function prepare_validation_error( array $error = [], array $data = [] )
 				( 'style' === $node->nodeName && ! $node->hasAttribute( 'amp-custom' ) && ! $node->hasAttribute( 'amp-keyframes' ) )
 			) {
 				$error['text'] = $node->textContent;
+
+				// Normalize string and number literals to prevent nonces, random numbers, and timestamps from generating endless number of validation errors.
+				$error['text'] = preg_replace(
+					'/"([^"\\\\]*+|\\\\(\\\\\\\\)*+.)+"/',
+					'__DOUBLE_QUOTED_STRING__',
+					$error['text']
+				);
+				$error['text'] = preg_replace(
+					'/\'([^\'\\\\]*+|\\\\(\\\\\\\\)*+.)+\'/',
+					'__SINGLE_QUOTED_STRING__',
+					$error['text']
+				);
+				$error['text'] = preg_replace(
+					'/(\b|-)\d+\.\d+\b/',
+					'__FLOAT__',
+					$error['text']
+				);
+				$error['text'] = preg_replace(
+					'/(\b|-)\d+\b/',
+					'__INT__',
+					$error['text']
+				);
```

Comment:
It would be more efficient to use only 1 `preg_replace()` call. See https://3v4l.org/dYfYn for an example.

```suggestion
				$error['text'] = preg_replace(
					[
						'/"([^"\\\\]*+|\\\\(\\\\\\\\)*+.)+"/',
						'/\'([^\'\\\\]*+|\\\\(\\\\\\\\)*+.)+\'/',
						'/(\b|-)\d+\.\d+\b/',
						'/(\b|-)\d+\b/',
					],
					[
						'__DOUBLE_QUOTED_STRING__',
						'__SINGLE_QUOTED_STRING__',
						'__FLOAT__',
						'__INT__',
					],
					$error['text']
				);
```

Reply:
Good. Done in 3d69b0bd2. I didn't apply your suggestion directly here because I had another fix in between (14d7e58ffc17da3a4e58618921cd7ab50a76a2c6).